standalone-metastore/metastore-common/src/main/thrift/hive_metastore.thrift

```diff
@@ -1861,6 +1861,19 @@ struct ScheduledQueryProgressInfo{
   4: optional string errorMessage,
 }
 
+struct GetFileListRequest {
+  1: optional string catName,
+  2: optional string dbName,
+  3: optional string tableName,
+  4: optional list<string> partVals,
```

Comment:
Does this mean the file list is retrieved on 1 partition at a time? It seems like we can have bulk retrieval for a given list of partition names?

Reply:
Yes, it can retrieve for 1 partition at a time. It can handle multi-level partitions as well. Do you think it would make sense to support returning file metadata for a list of partition values? It can lead to a huge amount of data sent over the network and the cache can be filled easily with just a few entries.

Reply:
I think there is a trade-off here. On larger tables with lots of partitions, doing multiple RPCs to the metastore for fetching the file-metadata one at a time not only is less efficient, it is likely that the ValidWriteIdList is updated for the table during the time and the cache hit ratio could go down. You are right about large data sent over network. In my experience the file-metadata which we are sending here is few hundred bytes per partition and the its not very large even for few thousands of the partition. If use a partitionNames list here in the request, clients can always do batching like requesting 1000 partitions at a time which would be more efficient.